x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorIndexService.java

```diff
@@ -0,0 +1,368 @@
+        client.admin().cluster().putPipeline(putPipelineRequest);
+        logger.info("Created pipeline" + ENT_SEARCH_INGESTION_PIPELINE_NAME);
+    }
+
+    private static void ensureInternalIndex(Client client, String index, XContentBuilder mappings, Settings settings, String alias) {
+        GetIndexRequest getIndexRequest = new GetIndexRequest().indices(CONNECTOR_CONCRETE_INDEX_NAME);
+        ThreadContext threadContext = client.threadPool().getThreadContext();
+        executeAsyncWithOrigin(threadContext, ENT_SEARCH_ORIGIN, getIndexRequest, new ActionListener<GetIndexResponse>() {
+            public void onResponse(GetIndexResponse getIndexResponse) {
+                logger.info("Found " + index + " index.");
+            }
+
+            public void onFailure(Exception e) {
+                final Throwable cause = ExceptionsHelper.unwrapCause(e);
+                if (cause instanceof ResourceNotFoundException) {
+                    logger.error("Index " + index + " not found.");
+                    createInternalIndex(client, index, mappings, settings, alias);
+                } else {
+                    logger.error("Error getting " + index + " index " + e.toString());
+                }
+            }
+        }, client.admin().indices()::getIndex);
+    }
+
+    private static void ensurePipeline(Client client) {
+        GetPipelineRequest getPipelineRequest = new GetPipelineRequest(ENT_SEARCH_INGESTION_PIPELINE_NAME);
+        ThreadContext threadContext = client.threadPool().getThreadContext();
+        executeAsyncWithOrigin(threadContext, ENT_SEARCH_ORIGIN, getPipelineRequest, new ActionListener<>() {
+            public void onResponse(GetPipelineResponse getPipelineResponse) {
+                if (getPipelineResponse.isFound() == false) {
+                    createPipeline(client);
+                } else {
+                    logger.info("Found " + ENT_SEARCH_INGESTION_PIPELINE_NAME + " pipeline.");
+                }
+            }
+
+            public void onFailure(Exception e) {
+                final Throwable cause = ExceptionsHelper.unwrapCause(e);
+                if (cause instanceof ResourceNotFoundException) {
+                    logger.error("Pipeline " + ENT_SEARCH_INGESTION_PIPELINE_NAME + " not found.");
+                    createPipeline(client);
+                } else {
+                    logger.error("Error getting " + ENT_SEARCH_INGESTION_PIPELINE_NAME + " index " + e.toString());
+                }
+            }
+        }, client.admin().cluster()::getPipeline);
+    }
+
+    private static Settings getConnectorsIndexSettings() {
+        return Settings.builder()
+            .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+            .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
+            .put(IndexMetadata.SETTING_AUTO_EXPAND_REPLICAS, "0-3")
+            .put(IndexMetadata.SETTING_INDEX_HIDDEN, true)
+            .put("index.refresh_interval", "1s")
```

Comment:
I don't see that we used `index.refresh_interval` or `NUMBER_OF_SHARDS` in our ent-search creation of this index. We're confident in these settings as opposed to continuing to use defaults?

Reply:
We did set them in Kibana. I'm not particularly hung up on either of them, we can definitely skip refresh interval (default is one second anyway). Number of shards I'm pretty confident in: we shouldn't want to split out the connectors index ever, and it shouldn't ever grow to a size where that would be necessary.